### PR TITLE
Implement github authentication helper

### PR DESCRIPTION
**Pull request by tripodsan**
see #380, see #579 

**NOTE**: The name the `testZAuthCmd.js` file is choosen in such a way that tests are executed after the replay tests in `testPublishCmd.js`. it seems impossible to re-init replay after nock
messed with the http request classes.

### CODE REVIEW

**Comment by trieloff** [comment]
This pull request **introduces 1 alert** when merging 97c902081d9d1f8e0950107923edc5a006dc92f0 into b338328701a9eaf4ef54f607a23103bd116c5189 - [view on LGTM.com](https://lgtm.com/projects/g/adobe/helix-cli/rev/pr-947ab66bc4ed50a0b61da7897fc9121c2a72493a)

**new alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: src/cli-util.js [hunk 1]
```diff
@@ -10,12 +10,81 @@
...
 
+const readline = require('readline');
+const chalk = require('chalk');
 const packagejson = require('../package.json');
 
 function getVersion() {
   return packagejson.version;
 }
 
+const spinnerFrames = process.platform === 'win32'
```
Comment (trieloff) [inline src/cli-util.js]: Is this really a platform dependency or a terminal dependency, i.e. we can assume UTF-8 support when `$TERM=xterm`?

Comment (tripodsan) [reply src/cli-util.js]: don't know :-) I copy pasted it from somewhere. but let me try it out....

Comment (tripodsan) [reply src/cli-util.js]: in command prompt, auth doesn't work at all:

```
C:\Users\tripod>hlx --version
0.13.6-pre.9

C:\Users\tripod>hlx auth
Cannot read property 'split' of null
```

in mingw, it works. even with the unicode.

Comment (tripodsan) [reply src/cli-util.js]: apparently this is a known problem. https://github.com/sindresorhus/figures
I'd stick to the distinction for know.

**Pull request merged**